synapse/lib/snap.py

```diff
@@ -1404,25 +1404,54 @@ async def _addGuidNodeByDict(self, form, vals, props=None):
 
         trycast = vals.pop('$try', False)
         addprops = vals.pop('$props', None)
-        if addprops is not None:
-            props.update(addprops)
 
-        try:
-            for name, valu in list(props.items()):
+        if not vals:
+            mesg = f'No values provided for form {form.full}'
+            raise s_exc.BadTypeValu(mesg=mesg)
+
+        for name, valu in list(props.items()):
+            try:
                 props[name] = form.reqProp(name).type.norm(valu)
+            except s_exc.BadTypeValu as e:
+                mesg = e.errinfo.get('mesg')
+                e.update({
+                    'prop': name,
+                    'form': form.name,
+                    'mesg': f'Bad value for prop {name}: {mesg}',
+                })
```

Comment:
Could any of these be already set? Should we use `e.setdefault()` instead for each key?

Reply:
None of the places I've found where we raise a BadTypeValu really set these (they set other keys like `valu`), and even if they did, I think right here since we're trying to norm that form and prop, it makes sense to have those set as the primary values?

Reply:
Sounds good to me. Thanks for checking.

Reply:
checked our current use of BadTypeValu - didn't see any uses of those keys. This is similar to other code we have in the snap already.